phylogenetic/rules/merge_inputs.smk

```diff
@@ -93,14 +85,82 @@ def _gather_inputs():
     return {el['name']: {k:(v if k=='name' else path_or_url(v)) for k,v in el.items()} for el in all_inputs}
 
 input_sources = _gather_inputs()
+_input_metadata = [info['metadata'] for info in input_sources.values() if info.get('metadata', None)]
+_input_sequences = [info['sequences'] for info in input_sources.values() if info.get('sequences', None)]
 
 def input_metadata(wildcards):
-    inputs = [info['metadata'] for info in input_sources.values() if info.get('metadata', None)]
-    return inputs[0] if len(inputs)==1 else rules.merge_metadata.output.metadata
+    return (
+        rules.decompress_metadata.output.metadata
+        if len(_input_metadata)==1
+        else rules.merge_metadata.output.metadata
+    )
 
 def input_sequences(wildcards):
-    inputs = [info['sequences'] for info in input_sources.values() if info.get('sequences', None)]
-    return inputs[0] if len(inputs)==1 else rules.merge_sequences.output.sequences
+    return (
+        rules.decompress_sequences.output.sequences
+        if len(_input_sequences)==1
+        else rules.merge_sequences.output.sequences
+    )
+
+
+def strip_compression_ext(input: str) -> str:
+    expected_compression_extensions = {
+        ".gz",
+        ".bz2",
+        ".xz",
+        ".zst",
+    }
+    input_path= Path(input)
+    return (
+        str(input_path.with_suffix(""))
+        if input_path.suffix in expected_compression_extensions
+        else input
+    )
+
+
+rule decompress_metadata:
+    """
+    This rule should only be invoked if there is a single metadata input to
+    ensure that we have a decompressed input for downstream rules to match
+    the output of rule.merge_metadata.
+    """
+    input:
+        metadata = _input_metadata[0],
+    output:
+        metadata = f"results/{remove_compression_ext(_input_metadata[0])}",
```

Comment:
_noticed in passing_

This calls `remove_compression_ext()` but the function above is `strip_compression_ext()`. (Ditto for the rule below.)